### PR TITLE
feat: Add button to hide thread upsell

### LINKED ISSUE #4531: Signed out 'join community' upsell on thread views covers most of the content
Ref: https://twitter.com/sergiodxa/status/1081745429681897472

Need to figure out how to approach this so that signed out users still have a quick way to sign-in and expose the chat input without having to scroll all the way to the bottom of chat first.

### PR DESCRIPTION
**Pull request by arthurdenner**
When the upsell is closed, the action related to it becomes available through a floating button.

**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hyperion (frontend)

**Related issues (delete if you don't know of any)**
Closes #4531.

### CODE REVIEW

**Comment by arthurdenner** [comment]
Do I need to add `// @flow` on `src/components/buttons/index.js` to get rid of the warning above?

**Comment by mxstbr** [comment]
@arthurdenner do not worry about it, you are solid! 👍 Going to defer to @brianlovin for a UI review here

File: src/components/upsell/joinChannel.js [hunk 1]
```diff
@@ -92,13 +95,31 @@ class JoinChannel extends React.Component<Props, State> {
...
-    const { isLoading } = this.state;
+    const { isLoading, isOpen } = this.state;
     const { channel, community, currentUser } = this.props;
 
+    if (!isOpen) {
+      return (
+        <FloatingButton
+          loading={isLoading}
+          onClick={!currentUser ? this.login : this.toggleSubscription}
+          icon={!currentUser ? 'door-enter' : 'plus'}
```
Comment (arthurdenner) [inline src/components/upsell/joinChannel.js]: Maybe we should add the `door-enter` icon to the `Login or sign up` button as well now.

**Comment by brianlovin** [comment]
@arthurdenner pulled this locally and started trying to make things feel good, but the more I worked on this the more I realized how much we can simplify this. Most of the content in that upsell is repetitive, and we're getting too weird with having to dismiss it. My last commit simplifies everything to just a floating button - want to pull that and take a look?

**Comment by arthurdenner** [comment]
Hey, @brianlovin, I'm glad you simplified the UI, it's way better now.

I'd just suggest to remove the `16px` padding applied on smaller screens in the `JoinChannelContainer` component, keeping it at `8px` looks better imo.

**Comment by brianlovin** [comment]
Agreed, good catch! Let's ship this :)

**Comment by brianlovin** [comment]
Thanks again for getting the momentum going here @arthurdenner  :)

**Pull request merged**